### ISSUE TITLE
Dataview comparison ignores offsets

### ISSUE THREAD

**Issue by demurgos** [chaijs/deep-eql#53]
The dataview comparison in `ExtensiveDeepEqualByType` compares the underlying buffers instead of comparing the dataviews themselves.
This means that the following dataviews are treated as different even if they represent the same data:

```javascript
const shortBuffer = new Uint8Array([0x01, 0x01]);
const longBuffer = new Uint8Array([0x00, 0x01, 0x02, 0x03]);
const view1 = new DataView(shortBuffer, 0, 1);
const view2 = new DataView(longBuffer, 1, 1);
``` 

Both views represent the slice `[0x01]` but the equality test returns false because the underlying buffers are different.
Note that this behavior is not even fully consistent. If you want to treat `buffer` as a public property that should match in both cases, then the offset should also matter. At the moment `new DataView(longBuffer, 0, 2)` is equal to `new DataView(2, 2)` because the buffers are the same (even if the data is different due to the different offsets).

**Comment by keithamus** [comment]
Seems like this would be useful. How do you propose we make this change? We could `.slice()` the ArrayBuffers by the respective `byteOffset` and `byteLength` and compare the slices values?

**Comment by demurgos** [comment]
Yes, I think that comparing the corresponding slices is the expected behavior in the case of deep equality.
